src/kde/kde.cc

```diff
@@ -0,0 +1,154 @@
+//--------------------------------------------*-C++-*---------------------------------------------//
+/*!
+ * \file   kde/kde.cc
+ * \author Mathew Cleveland
+ * \date   November 10th 2020
+ * \brief  Explicitly defined KDE functions for various dimensions and coordinate
```

Comment:
@alexrlongne Here is a more detailed description of what I've implemented. I made an effort to address some of the conversation that @timburculosis and I had about this approach and how it compares with what has been done in the past.

Reply:
Thanks Matt!